python/common/org/Python.java

```diff
@@ -915,10 +915,60 @@ public static void initializeModule(java.lang.Class cls, java.util.Map<java.lang
             "With a single iterable argument, return its biggest item. The\n" +
             "default keyword-only argument specifies an object to return if\n" +
             "the provided iterable is empty.\n" +
-            "With two or more arguments, return the largest argument.\n"
+            "With two or more arguments, return the largest argument.\n",
+        args = {"first"},
+        varargs = "others",
+        kwargs = "kwargs"
     )
-    public static org.python.Object max() {
-        throw new org.python.exceptions.NotImplementedError("Builtin function 'max' not implemented");
+    public static org.python.Object max(org.python.Object first, org.python.types.Tuple rest, org.python.types.Dict kwargs) {
+        org.python.Object val = null;
+        if (first == null) {
+            throw new org.python.exceptions.TypeError("max expected 1 arguments, got 0");
+        } else {
+            org.python.Iterable iter;
+            if (first instanceof org.python.types.List) {
```

Comment:
Ok, I tried iterating through the params, it's breaking at some places:
```python
max("Alpha", "Scotty", "Bones", "Spock", "Kirk")
- p
+ Spock
```

But working for similar cases:
```python
max("foo", "bar", "other")
other
```

I'm also not able to find out where a `java.util.NoSuchElementException` is produced, when an empty array is passed to as first.